mozilla/addons-server
# verify pip dependencies in CI

### Description

Add check to CI verifying that pip packages are installed in the container.

- add script to determine all dependencies specified in a set of requirements.txt files are installed via `pip show`
- add pip_development stage to docker build to install dev related dependencies (like dennis previously in it's own requirements file) (not included in final image)
- merge locale.txt into dev.txt dependencies (we have dev stage now, don't need it)
- add ci check to verify pip. packages are installed on PRs

### Context

We rely on some (mildly) complex logic to install our dependencies. For this reason, it makes sense to verify that dependencies that are "required" actually get installed in the final container. This PR adds such a check.

Having this check is very useful (for example) in this PR that wants to modify how dependencies are finally added to the image https://github.com/mozilla/addons-server/pull/22253

### Testing

You can verify this check works, by adding a package to a requirements.txt file, NOT installing it, and running the check command specifying that file.

1. add a package (any package) to prod.txt
2. run `make check`
3. should exit 1 that the package you added is "<package": Missing"
